tools/tf2openapi/cmd/e2e_test.go

```diff
@@ -0,0 +1,196 @@
+	cmd := exec.Command(cmdName, cmdArgs...)
+	spec, err := cmd.Output()
+	g.Expect(err).Should(gomega.BeNil())
+	swagger := &openapi3.Swagger{}
+	g.Expect(json.Unmarshal([]byte(spec), &swagger)).To(gomega.Succeed())
+
+	expectedSpec := readFile("TestCensus.golden", t)
+	expectedSwagger := &openapi3.Swagger{}
+	g.Expect(json.Unmarshal(expectedSpec, &expectedSwagger)).To(gomega.Succeed())
+
+	// test equality, ignoring order in JSON arrays
+	instances := swagger.Components.RequestBodies["modelInput"].Value.Content.Get("application/json").
+		Schema.Value.Properties["instances"].Value
+	expectedInstances := expectedSwagger.Components.RequestBodies["modelInput"].Value.Content.
+		Get("application/json").Schema.Value.Properties["instances"].Value
+	g.Expect(swagger.Paths).Should(gomega.Equal(expectedSwagger.Paths))
+	g.Expect(swagger.OpenAPI).Should(gomega.Equal(expectedSwagger.OpenAPI))
+	g.Expect(swagger.Info).Should(gomega.Equal(expectedSwagger.Info))
+	g.Expect(swagger.Components.Responses).Should(gomega.Equal(expectedSwagger.Components.Responses))
+	g.Expect(instances.Items.Value.Required).Should(gomega.Not(gomega.BeNil()))
+	g.Expect(instances.Items.Value.Required).To(gomega.ConsistOf(expectedInstances.Items.Value.Required))
+	g.Expect(instances.Items.Value.AdditionalPropertiesAllowed).Should(gomega.Not(gomega.BeNil()))
+	g.Expect(instances.Items.Value.AdditionalPropertiesAllowed).Should(gomega.Equal(expectedInstances.Items.Value.AdditionalPropertiesAllowed))
+	g.Expect(instances.Items.Value.Properties).Should(gomega.Equal(expectedInstances.Items.Value.Properties))
+}
+
+func TestInvalidFilePath(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	wd := workingDir(t)
+	cmdName := cmd(wd)
+	cmdArgs := []string{"--model_base_path", "badPath"}
+	cmd := exec.Command(cmdName, cmdArgs...)
+	output, err := cmd.CombinedOutput()
+	g.Expect(err.(*exec.ExitError).ExitCode()).To(gomega.Equal(1))
+	expectedErr := fmt.Sprintf(ModelBasePathError, "badPath", "")
+	g.Expect(output).Should(gomega.ContainSubstring(expectedErr))
+}
+
+func TestCustomFlags(t *testing.T) {
+	// estimator model src: https://github.com/GoogleCloudPlatform/cloudml-samples/tree/master/census
+	g := gomega.NewGomegaWithT(t)
+	wd := workingDir(t)
+	cmdName := cmd(wd)
+	cmdArgs := []string{"--model_base_path", wd + "/testdata/" + t.Name() + ".pb",
+		"--signature_def", "predict", "--name", "customName", "--version", "1000",
+		"--metagraph_tags", "serve"}
+	cmd := exec.Command(cmdName, cmdArgs...)
+	spec, err := cmd.Output()
+	g.Expect(err).Should(gomega.BeNil())
+	swagger := &openapi3.Swagger{}
+	g.Expect(json.Unmarshal([]byte(spec), &swagger)).To(gomega.Succeed())
+
+	expectedSpec := readFile("TestCustomFlags.golden", t)
+	expectedSwagger := &openapi3.Swagger{}
+	g.Expect(json.Unmarshal(expectedSpec, &expectedSwagger)).To(gomega.Succeed())
+
+	// test equality, ignoring order in JSON arrays
```

Comment:
Could these individual Expect statements be abstracted into a `expectJsonEquality(swagger, expectedSwagger)` method?

Then this method could be reused here and also in `TestCensus` above.

Reply:
done

Reply:
done